frontend/src/pages/auth/signup/SignUp.tsx

```diff
@@ -1,12 +1,156 @@
+            const notification = createNotification(Notifications.SIGN_UP, NotificationType.SUCCESS);
+            dispatch(showNotification(notification));
+        } else {
+            const notification = createNotification(Notifications.VALIDATION_NOT_PASSED, NotificationType.ERROR);
+            dispatch(showNotification(notification));
+        }
+    };
+
+    const updateEmailChange = (email: string, isEmailValid: boolean) => {
+        setUserCredentials({...userCredentials, email});
+        setFieldsValidation({...fieldsValidation, isEmailValid});
+    };
+
+    const updateNameChange = (name: string, isNameValid: boolean) => {
+        setUserCredentials({...userCredentials, name});
+        setFieldsValidation({...fieldsValidation, isNameValid});
+    };
+
+    const updatePasswordChange = (password: string, isPasswordValid: boolean) => {
+        setUserCredentials({...userCredentials, password});
+        setFieldsValidation({...fieldsValidation, isPasswordValid});
+    };
+
+    useEffect(() => {
+        setIsValid(
+            fieldsValidation.isEmailValid &&
+            fieldsValidation.isPasswordValid &&
+            fieldsValidation.isNameValid &&
+            selectedRole &&
+            agreeChecked
+        );
+    }, [fieldsValidation, selectedRole, agreeChecked]);
+
     return (
-        <p>
-            {Titles.SIGN_UP}
-        </p>
+        <Box className="auth-container">
+            <Box className="auth-content">
+                <AuthLogo/>
+                <EmailField
+                    defaultEmail={userCredentials.email}
+                    updateEmailFields={updateEmailChange}
+                />
+                <NameField
+                    defaultName={userCredentials.name}
+                    updateNameFields={updateNameChange}
+                />
+                <PasswordField
+                    defaultPassword={userCredentials.password}
+                    updatePasswordFields={updatePasswordChange}
+                />
+                <Typography variant="h5" gutterBottom>
+                    {Titles.SELECT_ROLES}
+                </Typography>
+                <Box className="role-selector">
+                    <Box className="role-option">
+                        <Radio
+                            checked={selectedRole === UserType.Customer}
+                            onChange={handleRoleChange}
+                            name="radio-buttons"
+                            value={UserType.Customer}
+                        />
+                        <Typography variant="body2">{Roles.CUSTOMER}</Typography>
+                    </Box>
+                    <Box className="role-option">
+                        <Radio
+                            checked={selectedRole === UserType.Merchant}
+                            onChange={handleRoleChange}
+                            name="radio-buttons"
+                            value={UserType.Merchant}
+                        />
```

Comment:
Why did you decide to use manual approach of creating separate radio buttons instead of automatic radio group ?
https://mui.com/material-ui/react-radio-button/#direction